portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,23 +32,46 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-  private final ArrayList<String> comments = new ArrayList<>();
     
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = new Gson().toJson(comments);
+    System.out.println("hi");
+    Query query = new Query("Comment").addSort("year", SortDirection.DESCENDING)
+                                      .addSort("month", SortDirection.DESCENDING)
+                                      .addSort("day", SortDirection.DESCENDING);
+
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
```

Comment:
Most of the times it's best to declare the variable type, in this case ArrayList, to be as general as possible. By general I mean, using the interface if possible and initializing the variable to a specific implementation of the interface.

`List<String>` comments = new ArrayList<String>();`

We follow this because its usually best to program to interfaces, rather than implementations. In doing so, we don't limit our code to be bound to a specific type, which could make code harder to change later. E.g. with List, you can also initialize to a LinkedList if that seems to suit your needs better.

Declaring to the specific implementation is also ok if we need to use some specific member of ArrayList that isn't in List. But I don't think we need to do that here so we should be good with declaring List. Hope this explains a little, but always down to chat/discuss more on this if you have any concerns

Reply:
Ah okay, done.